index.js

```diff
@@ -109,7 +109,10 @@ class TreeNode {
       c = b4a.compare(key.value, await this.getKey(mid))
 
       if (c === 0) {
-        if (cas && !(await cas((await this.getKeyNode(mid)).final(encoding), node))) return true
+        const older = (await this.getKeyNode(mid)).final(encoding)
```

Comment:
I think this change makes inserts slower for the case where no cas is defined, because it now always runs `getKeyNode()`, which I believe runs in logarithmic time. I'm not convinced that's a good trade-off to make for this feature--can't we special case the cas-case and keep the old code when no cas is specified?

Reply:
yea this needs to only run on cas

Reply:
just make an if

```js
if (cas) {
  const older = ....
  const swap = await cas(older, node)
  if (!swap || swap === older) return true
}
```

Reply:
if it swaps you need to update the value you insert also

Reply:
> if it swaps you need to update the value you insert also

Where am I missing this?